cmd/root.go

```diff
@@ -49,8 +51,14 @@ func init() {
 	// Here you will define your flags and configuration settings.
 	// Cobra supports persistent flags, which, if defined here,
 	// will be global for your application.
+	flags := rootCmd.PersistentFlags()
+	// Normalize all flags that are coming from other packages or pre-configurations
+	// a.k.a. change all "_" to "-". e.g. klog package
+	flags.SetNormalizeFunc(cliflag.WordSepNormalizeFunc)
 
-	rootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", fmt.Sprintf("config file (default is $HOME/%s/%s.yaml", gardenHomeFolder, configName))
+	addKlogFlags(flags)
+
+	flags.StringVar(&cfgFile, "config", "", fmt.Sprintf("config file (default is $HOME/%s/%s.yaml", gardenHomeFolder, configName))
```

Comment:
filepath.Join will also work on windows. $HOME is not correct in powershell or fish.